File: TOMIMSV4/Entity/Mantenimientos/Ajustes/clsBeTrans_ajuste_det.vb
' ***********************************************************************
' Assembly         : Entity
' Author           : ejcalderon
' Created          : 01-24-2018
'
' Last Modified By : ejcalderon
' Last Modified On : 01-26-2018
' ***********************************************************************
' <copyright file="clsBeTrans_ajuste_det.vb" company="TEAM OS">
'     Copyright ©  2016
' </copyright>
' <summary></summary>
' ***********************************************************************
''' <summary>
''' Class clsBeTrans_ajuste_det.
''' </summary>
''' <seealso cref="System.ICloneable" />
Public Class clsBeTrans_ajuste_det
    Implements ICloneable
    ''' <summary>
    ''' Gets or sets the idajustedet.
    ''' </summary>
    ''' <value>The idajustedet.</value>
    Public Property IdAjusteDet() As Integer = 0
    ''' <summary>
    ''' Gets or sets the idajusteenc.
    ''' </summary>
    ''' <value>The idajusteenc.</value>
    Public Property IdAjusteEnc() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier stock.
    ''' </summary>
    ''' <value>The identifier stock.</value>
    Public Property IdStock() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier propietario bodega.
    ''' </summary>
    ''' <value>The identifier propietario bodega.</value>
    Public Property IdPropietarioBodega() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier producto bodega.
    ''' </summary>
    ''' <value>The identifier producto bodega.</value>
    Public Property IdProductoBodega() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier producto estado.
    ''' </summary>
    ''' <value>The identifier producto estado.</value>
    Public Property IdProductoEstado() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier presentacion.
    ''' </summary>
    ''' <value>The identifier presentacion.</value>
    Public Property IdPresentacion() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier unidad medida.
    ''' </summary>
    ''' <value>The identifier unidad medida.</value>
    Public Property IdUnidadMedida() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier ubicacion.
    ''' </summary>
    ''' <value>The identifier ubicacion.</value>
    Public Property IdUbicacion() As Integer = 0
    ''' <summary>
    ''' Gets or sets the lote_original.
    ''' </summary>
    ''' <value>The lote_original.</value>
    Public Property Lote_original() As String = ""
    ''' <summary>
    ''' Gets or sets the lote_nuevo.
    ''' </summary>
    ''' <value>The lote_nuevo.</value>
    Public Property Lote_nuevo() As String = ""
    ''' <summary>
    ''' Gets or sets the fecha_vence_original.
    ''' </summary>
    ''' <value>The fecha_vence_original.</value>
    Public Property Fecha_vence_original() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets the fecha_vence_nueva.
    ''' </summary>
    ''' <value>The fecha_vence_nueva.</value>
    Public Property Fecha_vence_nueva() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets the peso_original.
    ''' </summary>
    ''' <value>The peso_original.</value>
    Public Property Peso_original() As Double = 0.0
    ''' <summary>
    ''' Gets or sets the peso_nuevo.
    ''' </summary>
    ''' <value>The peso_nuevo.</value>
    Public Property Peso_nuevo() As Double = 0.0
    ''' <summary>
    ''' Gets or sets the cantidad_original.
    ''' </summary>
    ''' <value>The cantidad_original.</value>
    Public Property Cantidad_original() As Double = 0.0
    ''' <summary>
    ''' Gets or sets the cantidad_nueva.
    ''' </summary>
    ''' <value>The cantidad_nueva.</value>
    Public Property Cantidad_nueva() As Double = 0.0
    ''' <summary>
    ''' Gets or sets the codigo_producto.
    ''' </summary>
    ''' <value>The codigo_producto.</value>
    Public Property Codigo_producto() As String = ""
    ''' <summary>
    ''' Gets or sets the nombre_producto.
    ''' </summary>
    ''' <value>The nombre_producto.</value>
    Public Property Nombre_producto() As String = ""
    ''' <summary>
    ''' Gets or sets the idtipoajuste.
    ''' </summary>
    ''' <value>The idtipoajuste.</value>
    Public Property Idtipoajuste() As Integer = 0
    ''' <summary>
    ''' Gets or sets the idmotivoajuste.
    ''' </summary>
    ''' <value>The idmotivoajuste.</value>
    Public Property IdMotivoAjuste() As Integer = 0
    ''' <summary>
    ''' Gets or sets the observacion.
    ''' </summary>
    ''' <value>The observacion.</value>
    Public Property Observacion() As String = ""
    ''' <summary>
    ''' Gets or sets the codigo_ajuste.
    ''' </summary>
    ''' <value>The codigo_ajuste.</value>
    Public Property Codigo_ajuste() As String = ""
    ''' <summary>
    ''' Gets or sets a value indicating whether this <see cref="clsBeTrans_ajuste_det"/> is enviado.
    ''' </summary>
    ''' <value><c>true</c> if enviado; otherwise, <c>false</c>.</value>
    Public Property Enviado() As Boolean = False
    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsBeTrans_ajuste_det"/> class.
    ''' </summary>

    Public Presentacion As New clsBeProducto_Presentacion

    Public Property IdBodegaERP As Integer = 0

    '#CKFK20220208 Agregamos el campo LicPlate
    Public Property lic_plate As String = ""

    '#CKFK20220208 Agregamos estos campos para control del envío de los ajustes a NAV
    Public Property referencia_ajuste_erp As String = ""
    Public Property estado_ajuste_erp As Boolean = False

    '#GT28082025: control de producto por talla/color
    Public Property IdProductoTallaColor_origen As Integer = 0
    Public Property Talla_origen As String = ""
    Public Property Color_origen As String = ""

    '#GT17122025: se agregan propiedades para poder dar traza si una talla cambio a otra, igual con color
    Public Property Talla_destino As String = ""
    Public Property Color_destino As String = ""
    Public Property IdProductoTallaColor_destino As Integer = 0

    Sub New()
    End Sub
    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsBeTrans_ajuste_det"/> class.
    ''' </summary>
    ''' <param name="idajustedet">The idajustedet.</param>
    ''' <param name="idajusteenc">The idajusteenc.</param>
    ''' <param name="IdStock">The identifier stock.</param>
    ''' <param name="IdPropietarioBodega">The identifier propietario bodega.</param>
    ''' <param name="IdProductoBodega">The identifier producto bodega.</param>
    ''' <param name="IdProductoEstado">The identifier producto estado.</param>
    ''' <param name="IdPresentacion">The identifier presentacion.</param>
    ''' <param name="IdUnidadMedida">The identifier unidad medida.</param>
    ''' <param name="IdUbicacion">The identifier ubicacion.</param>
    ''' <param name="lote_original">The lote_original.</param>
    ''' <param name="lote_nuevo">The lote_nuevo.</param>
    ''' <param name="fecha_vence_original">The fecha_vence_original.</param>
    ''' <param name="fecha_vence_nueva">The fecha_vence_nueva.</param>
    ''' <param name="peso_original">The peso_original.</param>
    ''' <param name="peso_nuevo">The peso_nuevo.</param>
    ''' <param name="cantidad_original">The cantidad_original.</param>
    ''' <param name="cantidad_nueva">The cantidad_nueva.</param>
    ''' <param name="codigo_producto">The codigo_producto.</param>
    ''' <param name="nombre_producto">The nombre_producto.</param>
    ''' <param name="idtipoajuste">The idtipoajuste.</param>
    ''' <param name="idmotivoajuste">The idmotivoajuste.</param>
    ''' <param name="observacion">The observacion.</param>
    ''' <param name="codigo_ajuste">The codigo_ajuste.</param>
    ''' <param name="enviado">if set to <c>true</c> [enviado].</param>
    Sub New(ByRef idajustedet As Integer, ByVal idajusteenc As Integer, ByVal IdStock As Integer, ByVal IdPropietarioBodega As Integer, ByVal IdProductoBodega As Integer, ByVal IdProductoEstado As Integer, ByVal IdPresentacion As Integer, ByVal IdUnidadMedida As Integer, ByVal IdUbicacion As Integer, ByVal lote_original As String, ByVal lote_nuevo As String, ByVal fecha_vence_original As Date, ByVal fecha_vence_nueva As Date, ByVal peso_original As Double, ByVal peso_nuevo As Double, ByVal cantidad_original As Double, ByVal cantidad_nueva As Double, ByVal codigo_producto As String, ByVal nombre_producto As String, ByVal idtipoajuste As Integer, ByVal idmotivoajuste As Integer, ByVal observacion As String, ByVal codigo_ajuste As String, ByVal enviado As Boolean)
        Me.IdAjusteDet = Idajustedet
        Me.IdAjusteEnc = Idajusteenc
        Me.IdStock = IdStock
        Me.IdPropietarioBodega = IdPropietarioBodega
        Me.IdProductoBodega = IdProductoBodega
        Me.IdProductoEstado = IdProductoEstado
        Me.IdPresentacion = IdPresentacion
        Me.IdUnidadMedida = IdUnidadMedida
        Me.IdUbicacion = IdUbicacion
        Me.Lote_original = Lote_original
        Me.Lote_nuevo = Lote_nuevo
        Me.Fecha_vence_original = Fecha_vence_original
        Me.Fecha_vence_nueva = Fecha_vence_nueva
        Me.Peso_original = Peso_original
        Me.Peso_nuevo = Peso_nuevo
        Me.Cantidad_original = Cantidad_original
        Me.Cantidad_nueva = Cantidad_nueva
        Me.Codigo_producto = Codigo_producto
        Me.Nombre_producto = Nombre_producto
        Me.Idtipoajuste = Idtipoajuste
        Me.IdMotivoAjuste = Idmotivoajuste
        Me.Observacion = Observacion
        Me.Codigo_ajuste = Codigo_ajuste
        Me.Enviado = Enviado
    End Sub
    ''' <summary>
    ''' Creates a new object that is a copy of the current instance.
    ''' </summary>
    ''' <returns>A new object that is a copy of this instance.</returns>
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class

File: WMS.EntityCore/Ajustes/clsBeTrans_ajuste_det.cs
using WMS.EntityCore.Producto;

namespace WMS.EntityCore.Ajustes
{    
    public partial class clsBeTrans_ajuste_det : ICloneable
    {
        // Propiedades existentes del partial
        public int idstockres { get; set; }
        public int idstocklink { get; set; } = 0;
        public int esnuevolink { get; set; } = 0;
        public string UmBas { get; set; } = "";
        public double CantReservada { get; set; }
        public double Factor { get; set; } = 0;
        public string Nombre_Presentacion { get; set; } = "";

        /// <summary>
        /// Gets or sets the idajustedet.
        /// </summary>
        /// <value>The idajustedet.</value>
        public int IdAjusteDet { get; set; } = 0;

        /// <summary>
        /// Gets or sets the idajusteenc.
        /// </summary>
        /// <value>The idajusteenc.</value>
        public int IdAjusteEnc { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier stock.
        /// </summary>
        /// <value>The identifier stock.</value>
        public int IdStock { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier propietario bodega.
        /// </summary>
        /// <value>The identifier propietario bodega.</value>
        public int IdPropietarioBodega { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier producto bodega.
        /// </summary>
        /// <value>The identifier producto bodega.</value>
        public int IdProductoBodega { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier producto estado.
        /// </summary>
        /// <value>The identifier producto estado.</value>
        public int IdProductoEstado { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier presentacion.
        /// </summary>
        /// <value>The identifier presentacion.</value>
        public int IdPresentacion { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier unidad medida.
        /// </summary>
        /// <value>The identifier unidad medida.</value>
        public int IdUnidadMedida { get; set; } = 0;

        /// <summary>
        /// Gets or sets the identifier ubicacion.
        /// </summary>
        /// <value>The identifier ubicacion.</value>
        public int IdUbicacion { get; set; } = 0;

        /// <summary>
        /// Gets or sets the lote_original.
        /// </summary>
        /// <value>The lote_original.</value>
        public string Lote_original { get; set; } = "";

        /// <summary>
        /// Gets or sets the lote_nuevo.
        /// </summary>
        /// <value>The lote_nuevo.</value>
        public string Lote_nuevo { get; set; } = "";

        /// <summary>
        /// Gets or sets the fecha_vence_original.
        /// </summary>
        /// <value>The fecha_vence_original.</value>
        public DateTime Fecha_vence_original { get; set; } = DateTime.Now;

        /// <summary>
        /// Gets or sets the fecha_vence_nueva.
        /// </summary>
        /// <value>The fecha_vence_nueva.</value>
        public DateTime Fecha_vence_nueva { get; set; } = DateTime.Now;

        /// <summary>
        /// Gets or sets the peso_original.
        /// </summary>
        /// <value>The peso_original.</value>
        public double Peso_original { get; set; } = 0.0;

        /// <summary>
        /// Gets or sets the peso_nuevo.
        /// </summary>
        /// <value>The peso_nuevo.</value>
        public double Peso_nuevo { get; set; } = 0.0;

        /// <summary>
        /// Gets or sets the cantidad_original.
        /// </summary>
        /// <value>The cantidad_original.</value>
        public double Cantidad_original { get; set; } = 0.0;

        /// <summary>
        /// Gets or sets the cantidad_nueva.
        /// </summary>
        /// <value>The cantidad_nueva.</value>
        public double Cantidad_nueva { get; set; } = 0.0;

        /// <summary>
        /// Gets or sets the codigo_producto.
        /// </summary>
        /// <value>The codigo_producto.</value>
        public string Codigo_producto { get; set; } = "";

        /// <summary>
        /// Gets or sets the nombre_producto.
        /// </summary>
        /// <value>The nombre_producto.</value>
        public string Nombre_producto { get; set; } = "";

        /// <summary>
        /// Gets or sets the idtipoajuste.
        /// </summary>
        /// <value>The idtipoajuste.</value>
        public int Idtipoajuste { get; set; } = 0;

        /// <summary>
        /// Gets or sets the idmotivoajuste.
        /// </summary>
        /// <value>The idmotivoajuste.</value>
        public int IdMotivoAjuste { get; set; } = 0;

        /// <summary>
        /// Gets or sets the observacion.
        /// </summary>
        /// <value>The observacion.</value>
        public string Observacion { get; set; } = "";

        /// <summary>
        /// Gets or sets the codigo_ajuste.
        /// </summary>
        /// <value>The codigo_ajuste.</value>
        public string Codigo_ajuste { get; set; } = "";

        /// <summary>
        /// Gets or sets a value indicating whether this <see cref="clsBeTrans_ajuste_det"/> is enviado.
        /// </summary>
        /// <value><c>true</c> if enviado; otherwise, <c>false</c>.</value>
        public bool Enviado { get; set; } = false;

        /// <summary>
        /// Gets or sets the presentacion.
        /// </summary>
        public clsBeProducto_presentacion Presentacion { get; set; } = new clsBeProducto_presentacion();

        /// <summary>
        /// Gets or sets the identifier bodega erp.
        /// </summary>
        public int IdBodegaERP { get; set; } = 0;

        /// <summary>
        /// CKFK20220208: Agregamos el campo LicPlate
        /// </summary>
        public string lic_plate { get; set; } = "";

        /// <summary>
        /// CKFK20220208: Agregamos estos campos para control del envío de los ajustes a NAV
        /// </summary>
        public string referencia_ajuste_erp { get; set; } = "";

        /// <summary>
        /// Gets or sets a value indicating whether estado_ajuste_erp.
        /// </summary>
        public bool estado_ajuste_erp { get; set; } = false;

        /// <summary>
        /// GT28082025: control de producto por talla/color
        /// </summary>
        public int IdProductoTallaColor_origen { get; set; } = 0;

        /// <summary>
        /// Gets or sets the talla_origen.
        /// </summary>
        public string Talla_origen { get; set; } = "";

        /// <summary>
        /// Gets or sets the color_origen.
        /// </summary>
        public string Color_origen { get; set; } = "";

        /// <summary>
        /// GT17122025: se agregan propiedades para poder dar traza si una talla cambio a otra, igual con color
        /// </summary>
        public string Talla_destino { get; set; } = "";

        /// <summary>
        /// Gets or sets the color_destino.
        /// </summary>
        public string Color_destino { get; set; } = "";

        /// <summary>
        /// Gets or sets the identifier producto talla color destino.
        /// </summary>
        public int IdProductoTallaColor_destino { get; set; } = 0;

        /// <summary>
        /// Initializes a new instance of the <see cref="clsBeTrans_ajuste_det"/> class.
        /// </summary>
        public clsBeTrans_ajuste_det()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="clsBeTrans_ajuste_det"/> class.
        /// </summary>
        /// <param name="idajustedet">The idajustedet.</param>
        /// <param name="idajusteenc">The idajusteenc.</param>
        /// <param name="IdStock">The identifier stock.</param>
        /// <param name="IdPropietarioBodega">The identifier propietario bodega.</param>
        /// <param name="IdProductoBodega">The identifier producto bodega.</param>
        /// <param name="IdProductoEstado">The identifier producto estado.</param>
        /// <param name="IdPresentacion">The identifier presentacion.</param>
        /// <param name="IdUnidadMedida">The identifier unidad medida.</param>
        /// <param name="IdUbicacion">The identifier ubicacion.</param>
        /// <param name="lote_original">The lote_original.</param>
        /// <param name="lote_nuevo">The lote_nuevo.</param>
        /// <param name="fecha_vence_original">The fecha_vence_original.</param>
        /// <param name="fecha_vence_nueva">The fecha_vence_nueva.</param>
        /// <param name="peso_original">The peso_original.</param>
        /// <param name="peso_nuevo">The peso_nuevo.</param>
        /// <param name="cantidad_original">The cantidad_original.</param>
        /// <param name="cantidad_nueva">The cantidad_nueva.</param>
        /// <param name="codigo_producto">The codigo_producto.</param>
        /// <param name="nombre_producto">The nombre_producto.</param>
        /// <param name="idtipoajuste">The idtipoajuste.</param>
        /// <param name="idmotivoajuste">The idmotivoajuste.</param>
        /// <param name="observacion">The observacion.</param>
        /// <param name="codigo_ajuste">The codigo_ajuste.</param>
        /// <param name="enviado">if set to <c>true</c> [enviado].</param>
        public clsBeTrans_ajuste_det(
            int idajustedet,
            int idajusteenc,
            int IdStock,
            int IdPropietarioBodega,
            int IdProductoBodega,
            int IdProductoEstado,
            int IdPresentacion,
            int IdUnidadMedida,
            int IdUbicacion,
            string lote_original,
            string lote_nuevo,
            DateTime fecha_vence_original,
            DateTime fecha_vence_nueva,
            double peso_original,
            double peso_nuevo,
            double cantidad_original,
            double cantidad_nueva,
            string codigo_producto,
            string nombre_producto,
            int idtipoajuste,
            int idmotivoajuste,
            string observacion,
            string codigo_ajuste,
            bool enviado)
        {
            IdAjusteDet = idajustedet;
            IdAjusteEnc = idajusteenc;
            this.IdStock = IdStock;
            this.IdPropietarioBodega = IdPropietarioBodega;
            this.IdProductoBodega = IdProductoBodega;
            this.IdProductoEstado = IdProductoEstado;
            this.IdPresentacion = IdPresentacion;
            this.IdUnidadMedida = IdUnidadMedida;
            this.IdUbicacion = IdUbicacion;
            Lote_original = lote_original ?? "";
            Lote_nuevo = lote_nuevo ?? "";
            Fecha_vence_original = fecha_vence_original;
            Fecha_vence_nueva = fecha_vence_nueva;
            Peso_original = peso_original;
            Peso_nuevo = peso_nuevo;
            Cantidad_original = cantidad_original;
            Cantidad_nueva = cantidad_nueva;
            Codigo_producto = codigo_producto ?? "";
            Nombre_producto = nombre_producto ?? "";
            Idtipoajuste = idtipoajuste;
            IdMotivoAjuste = idmotivoajuste;
            Observacion = observacion ?? "";
            Codigo_ajuste = codigo_ajuste ?? "";
            Enviado = enviado;
        }

        /// <summary>
        /// Creates a new object that is a copy of the current instance.
        /// </summary>
        /// <returns>A new object that is a copy of this instance.</returns>
        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}